modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/RaftGroupServiceImpl.java

```diff
@@ -546,6 +547,23 @@ else if (resp0.errorCode() == RaftError.EPERM.getNumber() ||
                         fut.completeExceptionally(
                             new RaftException(RaftError.forNumber(resp0.errorCode()), resp0.errorMsg()));
                 }
+                else if (resp instanceof RpcRequests.SMErrorResponse) {
+                    SMThrowable th = ((RpcRequests.SMErrorResponse)resp).error();
+                    if (th instanceof SMCompactedThrowable) {
+                        try {
+                            Throwable restoredTh = (Throwable)Class.forName(((SMCompactedThrowable)th).throwableClassName())
+                                .getConstructor(String.class)
+                                .newInstance(((SMCompactedThrowable)th).throwableMessage());
+
+                            fut.completeExceptionally(restoredTh);
+                        }
+                        catch (Exception e) {
```

Comment:
Can you write a warning here?
"The exception does not have a constructor with a string message, IgniteException will be use."